src/audio/audioProcessor.js

```diff
@@ -31,38 +31,64 @@ export default class AudioProcessor {
       this.audible.connect(this.splitter);
       this.splitter.connect(this.analyserL, 0);
       this.splitter.connect(this.analyserR, 1);
-    }
 
-    // Initialised once as typed arrays
-    // Used for webaudio API raw (time domain) samples. 0 -> 255
-    this.timeByteArray = new Uint8Array(this.fftSize);
-    this.timeByteArrayL = new Uint8Array(this.fftSize);
-    this.timeByteArrayR = new Uint8Array(this.fftSize);
+      // Initialised once as typed arrays
+      // Used for webaudio API raw (time domain) samples. 0 -> 255
+      this.timeByteArray = new Uint8Array(this.fftSize);
+      this.timeByteArrayL = new Uint8Array(this.fftSize);
+      this.timeByteArrayR = new Uint8Array(this.fftSize);
 
-    // Signed raw samples shifted to -128 -> 127
-    this.timeArray = new Int8Array(this.fftSize);
-    this.timeByteArraySignedL = new Int8Array(this.fftSize);
-    this.timeByteArraySignedR = new Int8Array(this.fftSize);
+      // Signed raw samples shifted to -128 -> 127
+      this.timeArray = new Int8Array(this.fftSize);
+      this.timeByteArraySignedL = new Int8Array(this.fftSize);
+      this.timeByteArraySignedR = new Int8Array(this.fftSize);
 
-    // Undersampled from this.fftSize to this.numSamps
-    this.timeArrayL = new Int8Array(this.numSamps);
-    this.timeArrayR = new Int8Array(this.numSamps);
-  }
+      // Undersampled from this.fftSize to this.numSamps
+      this.timeArrayL = new Int8Array(this.numSamps);
+      this.timeArrayR = new Int8Array(this.numSamps);
 
+
+      this.freqArray = new Float32Array(this.numSamps);
+      this.freqArrayL = new Float32Array(this.numSamps);
+      this.freqArrayR = new Float32Array(this.numSamps);
+    }
+  }
+  /* eslint-disable no-bitwise */
   sampleAudio () {
     this.analyser.getByteTimeDomainData(this.timeByteArray);
     this.analyserL.getByteTimeDomainData(this.timeByteArrayL);
     this.analyserR.getByteTimeDomainData(this.timeByteArrayR);
-    this.processAudio();
+
+    this.analyser.getFloatFrequencyData(this.freqArray);
```

Comment:
whichever version of the FFT we use, I will want to use the same version in both the sampling scenario and the replaying audio via updateAudio scenario. So I think it would be better to just do things the way you did before, where `processAudio` gets called in both scenarios